pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -477,6 +477,15 @@ func (rd *restoreDataProcessor) processRestoreSpanEntry(
 		if log.V(5) {
 			log.Infof(ctx, "Put %s -> %s", key.Key, value.PrettyPrint())
 		}
+		// TODO(msbutler): figure out what import job ID to encode in MVCC Keys that
+		// are part of an in progress IMPORT. The central question: do we assign a
+		// new import job ID to the in progress IMPORT? If so, could we do that
+		// before ingestion, and write the new job ID here?
+		//
+		// Further, I don't think keys from a completed import need to preserve
+		// their import job ID. Ideally, this could be resolved on the backup side:
```

Comment:
Hmmm, interesting. I wonder if we can get away with not writing the import IDs in the general case her. This is speculative, but bear with me.

The only reason we need these import IDs is to handle restore at current timestamp. During backup, we know which imports are in flight. We can also know their start time. Could we add on the import IDs during export? In fact, if there can only be 1 import in flight per table, is it enough with a boolean (latter may have issues with chained backup/restores)?

Reply:
aha! I really like the idea of _only_ writing the ImportID during Export Request (or any other kv request used during streaming). While you're right that only 1 import can be in flight per table, we can't merely write a bool to the metadata because of incremental backups. Consider the following situation:

t0: Import A begins on foo
t1: Full backup of foo begins
t2: Import A ends 
t3: Import B begins on foo
t4: incremental backup of foo begins
t5: restore, only import B is still in progress

Suppose during backup, we only wrote an `IsImporting` bool to each key in the backup SSTs: the full backup would write that bool to some keys from import A, and the incremental backup would write the bool to some keys from import B. During restore, however, we'd have no way to distinguish between keys written during Import A vs B. So, each backed up, imported key, needs a unique ID. An easy unique ID is the job ID, but if we wanted to, we could potentially get away with writing a unique `short` to each key during Export. 

I.e.
- During import bind the shortID and the JobId to each table descriptor
- During backup planning:  gather all the unique shortIDs from active imports
- During export: write the shortID to each importing key
- During restore: create a map[oldShortID]NewShortID`, rewrite the restored descriptor and import job with the newShortID.

Reply:
Yep, makes sense. I don't think it's necessarily worth shortening the IDs, but if we could get away with only writing them for the rare cases where we're backing up in-flight imports, the savings would probably be worth it in the typical case.

Wdyt @dt?

Reply:
hmm. During an unrelated meeting, David pointed out the sad fact that our current approach of _only_ writing the ImportID during an Export Request falls apart for Backups/Restore's _of_ tenants, conducted by the system tenant. In this case, the system tenant has no access to the tenant's sql descriptors, and consequently, its importing job Ids. Therefore, there's currently no way to plumb the tenant's importing job IDs to the export request. I wonder though, the system tenant is so powerful! Is there an easy way to gather tenant descriptors during backup planning?

That being said, I think the original plan I proposed still works for tenant backup/restores, since the jobs table is completely preserved. This implies that there's no need to track and rewrite importing jobIDs for tenants, as there's no chance of jobID collision between the backed up and restored tenant.

Reply:
Aw, that's a bummer. Ok, back to the original plan then, I suppose.

Reply:
Well, we could just add some range state:
before you start importing, you send an rpc to the span you will import into saying the time above which you'll be importing and your import ID. distsender fans this out to every range in that span, and each of them persists this mapping in the range state. During splits it is duplicated to both sides, etc just like any other range state.
When we export eval, we can read the range state and apply the ID to any key with ts above the assigned value. Or we could just return the mapping from export at that point, instead of doing anything key-by-key.

I mean, this sorta sounds like range keys?

Reply:
also sounds a lot like our lock-span-for-op plans

Reply:
Hmm, that's an interesting idea. It's definitely doable, although we've currently hardcoded an assumption that MVCC range keys are range tombstones throughout the entire system since we didn't know what live range keys would look like yet. Probably a bit late in the game to change that with a week left before stability (and a ton of other work to do before then), even though I don't think it should be a huge lift to do something like that later.

Reply:
Then, when the import job is synthesized during RESTORE, this range state operation would happen again with the new job ID.

Reply:
GIven meeting yesterday, here's the [new approach](https://github.com/cockroachdb/cockroach/pull/85138#issuecomment-1205780978).